nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/RemoteProcessGroupResource.java

```diff
@@ -434,6 +436,197 @@ public Response updateRemoteProcessGroupOutputPort(
+                    @ApiResponse(code = 404, message = "The specified resource could not be found."),
+                    @ApiResponse(code = 409, message = "The request was valid but NiFi was not in the appropriate state to process it. Retrying the same request later may be successful.")
+            }
+    )
+    public Response updateRemoteProcessGroupInputPortRunStatus(
+            @Context final HttpServletRequest httpServletRequest,
+            @ApiParam(
+                    value = "The remote process group id.",
+                    required = true
+            )
+            @PathParam("id") final String id,
+            @ApiParam(
+                    value = "The remote process group port id.",
+                    required = true
+            )
+            @PathParam("port-id") final String portId,
+            @ApiParam(
+                    value = "The remote process group port.",
+                    required = true
+            ) final RemotePortRunStatusEntity requestRemotePortRunStatusEntity) {
+
+        if (requestRemotePortRunStatusEntity == null) {
+            throw new IllegalArgumentException("Remote process group port run status must be specified.");
+        }
+
+        if (requestRemotePortRunStatusEntity.getRevision() == null) {
+            throw new IllegalArgumentException("Revision must be specified.");
+        }
+
+        requestRemotePortRunStatusEntity.validateState();
+
+        if (isReplicateRequest()) {
+            return replicate(HttpMethod.PUT, requestRemotePortRunStatusEntity);
+        } else if (isDisconnectedFromCluster()) {
+            verifyDisconnectedNodeModification(requestRemotePortRunStatusEntity.isDisconnectedNodeAcknowledged());
+        }
+
+        final Revision requestRevision = getRevision(requestRemotePortRunStatusEntity.getRevision(), id);
+        final RemoteProcessGroupPortDTO remoteProcessGroupPort = new RemoteProcessGroupPortDTO();
+        remoteProcessGroupPort.setId(portId);
+        remoteProcessGroupPort.setGroupId(id);
+        remoteProcessGroupPort.setTransmitting(shouldTransmit(requestRemotePortRunStatusEntity));
+
+        return withWriteLock(
+                serviceFacade,
+                requestRemotePortRunStatusEntity,
+                requestRevision,
+                lookup -> {
+                    final Authorizable remoteProcessGroup = lookup.getRemoteProcessGroup(id);
+                    OperationAuthorizable.isAuthorized(remoteProcessGroup, authorizer, RequestAction.WRITE, NiFiUserUtils.getNiFiUser());
+                },
+                () -> serviceFacade.verifyUpdateRemoteProcessGroupInputPort(id, remoteProcessGroupPort),
+                (revision, remoteProcessGroupPortEntity) -> {
+                    // update the specified remote process group
+                    final RemoteProcessGroupPortEntity controllerResponse = serviceFacade.updateRemoteProcessGroupInputPort(revision, id, remoteProcessGroupPort);
```

Comment:
We need to recreate this `remoteProcessGroupPort` using the `remoteProcessGroupPortEntity` due to how we authorize/cache requests during our two phase commit.

Reply:
Fixed.